src/jit/lowerxarch.cpp

```diff
@@ -1030,6 +1034,23 @@ void Lowering::TreeNodeInfoInitShiftRotate(GenTree* tree)
     GenTreePtr shiftBy = tree->gtOp.gtOp2;
     GenTreePtr source  = tree->gtOp.gtOp1;
 
+#ifdef _TARGET_X86_
+    // The first operand of a GT_LSH_HI and GT_RSH_LO oper is a GT_LONG so that
+    // we can have a three operand form. Increment the srcCount.
+    if (source->OperGet() == GT_LONG)
+    {
+        info->srcCount++;
+        source->gtLsraInfo.isDelayFree = true;
```

Comment:
@sivarv I was actually wondering about if this is necessary. Not only was I marking both hi and lo as isDelayFree, but I've marked the GT_LONG node as isDelayFree. I just want to make sure this is necessary, or remove it if it isn't.